kategory/src/main/kotlin/kategory/data/StateT.kt

```diff
@@ -16,8 +16,8 @@ class StateT<F, S, A>(
     class F private constructor()
 
     companion object {
-        inline operator fun <reified F, S, A> invoke(run: StateTFunKind<F, S, A>, MF: Monad<F> = monad<F>()): StateT<F, S, A> =
-                StateT(MF, run)
+        operator fun <F, S, A> invoke(run: (S) -> HK<F, Tuple2<S, A>>, MF: Monad<F>): StateT<F, S, A> =
```

Comment:
For StateTFunKind, if it's not going to be used does it make sense to just remove the typealias too?